[Bhudisanont Kanit] iP

JumpyJay opened the pull request:
# Milo
> Tracking task made __simple__ (_and ~~fast~~ **REALLY FAST**_ 💨 💨 ) 
- [ ] like ticking this checkbox  ✅ 

With milo you could 
- Track to-do task 🏙️  
- Track deadlines 🌃 
- Track events 🌠 
- And more ❗  
.
.

To get started, all you have to do is
1. Download it from [here](https://github.com/JumpyJay/ip/releases/tag/A-Jar)
2. Start up the Jar file
3. Let Milo `do` the rest 😸😸😸

```
function tryNow() {
  console.log("Give it a go—it's absolutely free!");
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

JumpyJay committed:
Modify Duke.java

Skeletal implementation that greets the user.

Rename the bot's name from Duke to Bob- changed file name and class name

JumpyJay committed:
Modified Bob.java

Bob now echoes user commands after formatting it

Bob also now exits after the bye command is given

JumpyJay committed:
Modify Bob.java

Remove logo string variable

Implemented a switch to better check for input- would be useful when there are more cases

Added a list to store user text input

Created a separate printList method inside the Bob class

JumpyJay committed:
customise greeting and bye message

JumpyJay committed:
Implement Task object

JumpyJay committed:
Change bot name from Bob to Milo

JumpyJay committed:
Modify Milo.java

Split input into parts, with whitespace as breakpoints

Included new switch case, namely mark and unmark; allowing for marking and unmarking of completion status

Implement new methods (printMark and printUnmark) for the new switch cases

JumpyJay committed:
Add different types of task

3 new classes that inherits from Task, namely: Todos, Events, Deadlines

3 new switch case to support these class

New method to print when task is added

JumpyJay committed:
Modify Milo.java to handle errors

JumpyJay committed:
Modidy Milo.java to allow delete

Add case for "delete"

Create method to printDelete

Implement an ArrayList to store task data

JumpyJay committed:
Change print error message for enum switch

JumpyJay committed:
Test input against output

Received ' dos2unix: command not found' command even after changing the respective path and filenames

JumpyJay committed:
Abstract Printer and TaskType

JumpyJay committed:
Add Saver

Implement a saving system by making use of MiloData.txt

Modify Task, Todos, Deadlines, Events object to allow self-converting to text storing format

JumpyJay committed:
Merge branch 'branch-Level-7'

JumpyJay committed:
Add date

JumpyJay committed:
Merge branch 'branch-Level-8'

JumpyJay committed:
Add more OOP

JumpyJay committed:
Merge branch 'branch-A-MoreOOP'

JumpyJay committed:
Organise into Packages

Minor bug fix with the 'bye' command

JumpyJay committed:
Merge branch 'branch-A-Packages'

JumpyJay committed:
Merge branch 'add-gradle-support'

JumpyJay committed:
Add JUnit test

Same package but test could not be found

JumpyJay committed:
Add JacaDoc comments

JumpyJay committed:
Modify to follow coding standard

JumpyJay committed:
Implement Find feature

JumpyJay committed:
Merge tag 'A-JavaDoc'

JumpyJay committed:
Merge tag 'A-CodingStandard'

JumpyJay committed:
Merge Level-9 branch

JumpyJay committed:
Fix merge conflicts

Edwin1022 commented on src/test/java/Milo/MiloTest.java:
```diff
@@ -0,0 +1,40 @@
+
+public class MiloTest {
+    @Test
+    public void testGreetingAndExit() {
```
This works, but you could also try something like greetingAndExit_success?

Edwin1022 commented on src/test/java/Milo/UiTest.java:
```diff
@@ -0,0 +1,26 @@
+
+class UiTest {
+    @Test
+    public void testGreetUser() {
```
This works, but you could also try something like greetUser_success?

Edwin1022 commented on src/main/java/Milo/TaskObj/TaskType.java:
```diff
@@ -0,0 +1,15 @@
+ * Represents the different possible task type of the task object
+ */
+public class TaskType {
+    public static enum taskType {
```
I like how you use `enum` to represent all the task types.

Edwin1022 commented on src/main/java/Milo/TaskList.java:
```diff
@@ -0,0 +1,28 @@
+ */
+public class TaskList {
+
+    private final ArrayList<Task> todoList;
```
Would it be better to just name your ArrayList<Task> attribute here as taskList because there is a task type named Todos but taskList stores other tasks like Deadline tasks and Event tasks as well?

Edwin1022 commented on src/main/java/Milo/Ui.java:
```diff
@@ -0,0 +1,165 @@
+                ( ̳• · •̳)
+                /    づ♡
+                """;
+    private static final String hLine = "____________________________________________________________\n";
```
This works, but you could also try something like separationLine?

cweijin commented on src/main/java/Milo/Parser.java:
```diff
@@ -0,0 +1,133 @@
+    public static void readInput(String userInput, ArrayList<Task> todoList) {
+        String[] arrOfInput = userInput.split(" ", 2);
+        String action = arrOfInput[0];
+        switch (action) {
+            // Show list
+            case "list":
+                Ui.printList(todoList);
+                break;
+            // Find task
+            case "find":
```
I like how you use switch statement on the action keyword, but do check that your switch statement follows the coding convention (no indentation for case clauses).

cweijin commented on src/main/java/Milo/Parser.java:
```diff
@@ -0,0 +1,133 @@
+                    // Check case where event command is not properly formatted
+                    String[] eventDesc = arrOfInput[1].split("/from | /to", 3);
+                    if (eventDesc.length != 3) {
+                        Ui.printError(TaskType.taskType.DEADLINE, "Invalid event command\n Proper formatting: deadline <task description> + /from + <starting date description> + /to + <ending date description");
```
Perhaps can use line wrapping? (Line length should be no longer than 120 chars)

cweijin commented on src/main/java/Milo/Ui.java:
```diff
@@ -0,0 +1,165 @@
+    * @param error description
+     */
+
+    public static void printError(TaskType.taskType tasktype, String desc) {
+        String oops = "OOPS!!! ";
+        switch (tasktype) {
+            case TODO, EVENT, DEADLINE:
+                System.out.println(hLine + oops + desc + "\n" + hLine);
+                break;
+            case INVALID:
+                String invalidMessage = oops + "I'm sorry, but I don't know what that means ;-;\n";
+                System.out.println(hLine + invalidMessage + hLine);
+                break;
+            case DATE:
+                String invalidDateMessage = oops + "date is not properly formatted\n " +
+                        "It should be formatted as follow: YYYY-MM-DD\n";
+                System.out.println(hLine + invalidDateMessage + hLine);
+                break;
+            default:
+                String defaultMessage = oops + "I'm sorry, I don't know what that means ;-;\n";
+                System.out.println(hLine + defaultMessage + hLine);
+        }
+    }
+}
```
I like how you have a method to handle error printing.

cweijin commented on src/main/java/Milo/TaskObj/TaskType.java:
```diff
@@ -0,0 +1,15 @@
+/*
+ * Represents the different possible task type of the task object
+ */
+public class TaskType {
```
Is there any reason why you wrap the `enum` inside the `TaskType` class instead of having just `public enum TaskType`?

cweijin commented on src/main/java/Milo/TaskObj/TaskType.java:
```diff
@@ -0,0 +1,15 @@
+ */
+public class TaskType {
+    public static enum taskType {
+        TODO,
+        EVENT,
+        DEADLINE,
+        INVALID,
+
+        DATE
```
Reading through your code it seems like these are used to tag error to be handled by `printError` in Ui. Perhaps making it `ErrorType` instead of `TaskType`?

cweijin commented on src/main/java/Milo/Ui.java:
```diff
@@ -0,0 +1,165 @@
+ */
+public class Ui {
+    private final Scanner myScanner = new Scanner(System.in);
+    private final String cat0 = """
+                  ╱|、
+                (˚ˎ 。7 \s
+                 |、˜〵         \s
+                 じしˍ,)ノ
+                """;
+    private final String cat1 = """
+                 ∧,,,∧
+                ( ̳• · •̳)
+                /    づ♡
+                """;
```
So cute!! 😻

cweijin reviewed (commented):
LGTM! 👍 
Just some coding convention to follow.

JumpyJay committed:
Follow OOP principles more

JumpyJay committed:
Use CheckStyle

JumpyJay committed:
Add GUI

JumpyJay committed:
Merge branch 'branch-Level-10'

JumpyJay committed:
Abstract out Command

All the command was bunched together in Parser's readInput method, it
made the method exceed 70 lines of code hence does not have good code
quality

Split all the different types of command into their own respective
class

Thisis to improve code quality by avoid long methods and better adhere
to the SLAP principle

JumpyJay committed:
Improve code quality.

There were many unneccesary and unoptimised code present in the
last commit due to the drastic change of abstracting command in
the readInput method of the Parser class.

Implemented more inline variable, removed unused imports, removed
the unused fields in Ui class and minimise scope of variables,
specifically in the Ui class.

This is to improve the code quality, according to W5.4

JumpyJay committed:
Reduce variable scope

JumpyJay committed:
Add asserts

JumpyJay committed:
Merge pull request #1 from JumpyJay/branch-A-Assertions

Add asserts

JumpyJay committed:
Improve code quality

Major improvements have been done in the previous part
including abstracting methods to better align with the
SLAP principle

JumpyJay committed:
Improve code quality

JumpyJay committed:
Merge pull request #2 from JumpyJay/branch-A-CodeQuality

Improve code quality

JumpyJay committed:
Add client extension

JumpyJay committed:
Merge pull request #3 from JumpyJay/branch-D-Clients

Add client extension

JumpyJay committed:
Change UI

JumpyJay committed:
Merge branch 'branch-A-Personality'

JumpyJay committed:
Update naming

JumpyJay committed:
Update README.md

JumpyJay committed:
Update README.md

JumpyJay committed:
Update README.md

JumpyJay committed:
Update README.md

JumpyJay committed:
Update README.md

JumpyJay committed:
Update README.md

JumpyJay committed:
Update README.md

JumpyJay committed:
Update README.md

JumpyJay committed:
Update REAME.md
